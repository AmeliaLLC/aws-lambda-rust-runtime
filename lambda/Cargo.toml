[package]
name = "lambda"
version = "0.1.0"
authors = ["David Barsky <dbarsky@amazon.com>"]
edition = "2018"

[dependencies]
futures-preview = { version = "0.3.0-alpha.17", features = ["async-await", "nightly"] }
tokio = "0.2.0-alpha.1"
http = "0.1.17"
serde = { version = "1.0.91", features = ["derive"] }
serde_json = "1.0.39"
hyper = { git = "https://github.com/hyperium/hyper" }
proptest = "0.9.3"
headers = "0.2.1"
lazy_static = "1.3.0"
bytes = "0.4.12"
lambda-attributes = { path = "../lambda-attributes", version = "0.1.0" }

[dev-dependencies]
trybuild = "1"
