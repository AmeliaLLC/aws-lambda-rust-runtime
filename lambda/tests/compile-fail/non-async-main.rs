use lambda::lambda;

#[lambda]
#[tokio::main]
fn main() {}
