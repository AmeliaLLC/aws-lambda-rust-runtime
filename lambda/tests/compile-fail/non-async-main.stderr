error: the async keyword is missing from the function declaration
 --> $DIR/non-async-main.rs:3:1
  |
3 | #[lambda]
  | ^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
  |
  = note: consider adding a `main` function to `$DIR/tests/compile-fail/non-async-main.rs`

For more information about this error, try `rustc --explain E0601`.
error: Could not compile `lambda-tests`.
